packages/contracts/contracts/minter-suite/Minters/MinterSetPriceV5Holder.sol

```diff
@@ -0,0 +1,795 @@
+ * - manuallyLimitProjectMaxInvocations
+ * ----------------------------------------------------------------------------
+ * Additional admin and artist privileged roles may be described on other
+ * contracts that this minter integrates with.
+ * ----------------------------------------------------------------------------
+ * This contract allows gated minting with support for vaults to delegate minting
+ * privileges via an external delegation registry. This means a vault holding an
+ * allowed token can delegate minting privileges to a wallet that is not holding an
+ * allowed token, enabling the vault to remain air-gapped while still allowing minting.
+ * The delegation registry contract is responsible for managing these delegations,
+ * and is available at the address returned by the public immutable
+ * `delegationRegistryAddress`. At the time of writing, the delegation
+ * registry enables easy delegation configuring at https://delegate.cash/.
+ * Art Blocks does not guarentee the security of the delegation registry, and
+ * users should take care to ensure that the delegation registry is secure.
+ * Delegations must be configured by the vault owner prior to purchase. Supported
+ * delegation types include token-level, contract-level (via genArt721CoreAddress), or
+ * wallet-level delegation. Contract-level delegations must be configured for the core
+ * token contract as returned by the public immutable variable `genArt721CoreAddress`.
+ */
+contract MinterSetPriceV5Holder is
+    ReentrancyGuard,
+    ISharedMinterV0,
+    IFilteredSharedHolder
+{
+    // add Enumerable Set methods
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// Minter filter address this minter interacts with
+    address public immutable minterFilterAddress;
+
+    /// Minter filter this minter may interact with.
+    IMinterFilterV1 private immutable minterFilter;
+
+    /// Delegation registry address
+    address public immutable delegationRegistryAddress;
+
+    /// Delegation registry address
+    IDelegationRegistry private immutable delegationRegistryContract;
+
+    /// minterType for this minter
+    string public constant minterType = "MinterHolderV4";
+
+    /// minter version for this minter
+    string public constant minterVersion = "v4.1.0";
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    /// contractAddress => projectId => base project config
+    mapping(address => mapping(uint256 => ProjectConfig)) public projectConfig;
+
+    // STATE VARIABLES FOR SplitFundsLib
+    // contractAddress => IsEngineCache
+    mapping(address => SplitFundsLib.IsEngineCache) private _isEngineCaches;
+
+    // STATE VARIABLES FOR MaxInvocationsLib
+    /// contractAddress => projectId => max invocations specific project config
+    mapping(address => mapping(uint256 => MaxInvocationsLib.MaxInvocationsProjectConfig))
+        public maxInvocationsProjectConfig;
+
+    /// Set of core contracts allowed to be queried for token holders
```

Comment:
```suggestion
    // STATE VARIABLES FOR TokenHolderLib
    /// Set of core contracts allowed to be queried for token holders
```

Reply:
In addition to this suggestion, would probably suggest that we use a pattern of something like:

```
// STATE VARIABLES FOR {LibraryName}Lib begin here

...

// STATE VARIABLES FOR {LibraryName}Lib end here
```

Here and throughout such uses.